src/Symfony/Component/Console/Output/ConsoleSectionOutput.php

```diff
@@ -122,8 +122,9 @@ public function overwrite($message)
         $message = $this->getFormatter()->format($message);
 
         // make sure a non-empty message ends with a newline
-        if ($message !== '' && substr($message, -1) !== PHP_EOL) {
-            $message .= PHP_EOL;
+        $messageWithNewline = $message;
+        if ($messageWithNewline !== '' && substr($messageWithNewline, -1) !== PHP_EOL) {
```

Comment:
Please note, that `PHP_EOL` can be two characters wide (on Windows): https://heap.space/xref/php-src/main/php.h?r=bf29ee69#66

Hence, something like `str_ends_with($lastLine, \PHP_EOL)?` (From php 8.0 or polifill) would be required here.